doc/source/language/flow.rst

```diff
@@ -384,105 +386,324 @@ Halts execution for a specified amount of time, or until a specific set of crite
+
+    ON AG1 {
+      PRINT "You pressed '1', causing action group 1 to toggle.".
+      PRINT "Action group 1 is now " + AG1.
+      PRINT "No longer paying attention.".
+    }
+
+    WAIT UNTIL FALSE. // Wait forever.  You have to end with Control-C
+                      // The trigger will interrupt this waiting when it
+                      // notices it should.
 
-``ON``
-------
+For either ``WHEN`` or ``ON`` triggers, the check to see if it's
+time to trigger, and the subsequent run of the statements if they
+do trigger, interrupts the normal flow of the program.  The normal
+program flow will continue from where it left off, after the trigger
+finishes its work.
 
-The ``ON`` command is almost identical to the ``WHEN``/``THEN`` command. ``ON`` sets up a trigger in the background that will run the selected command exactly once when the boolean variable changes state from true to false or from false to true. This command is best used to listen for action group activations.
+In a sense, a trigger is a bit like a user function you created and
+then asked the kOS system to please keep running it again and again
+in the background until it finally says that it fired off.  In fact,
+it *is* implemented much like your own user functions.
 
-Just like with the ``WHEN``/``THEN`` command, the ``PRESERVE`` command can be used inside the code block to cause the trigger to remain active and not go away.
+If you run the above examples, you will see that they actually only
+happen once, and then stop happening again.  In the ON AG1 example,
+it will only fire off once, no matter how many times you press the
+'1' key.  More will be covered about how to change that further down.
 
 .. warning::
-    With the advent of :ref:`local variable scoping <scope>` in kOS
+    With the advent of :ref:`local variable scoping <trigger_scope>` in kOS
     version 0.17 and above, it's important to note that the variables
     used within the expression of a WHEN or an ON statement should
     be GLOBAL variables or the results are unpredictable.  If local
     variables were used, the results could change depending on where
     you are within the execution at the time.  
 
-How does it differ from ``WHEN``/``THEN``? The ``WHEN``/``THEN`` triggers are executed whenever the conditional expression *becomes true*. ``ON`` triggers are executed whenever the boolean variable *changes state* either from false to true or from true to false.
+.. warning::
+    Do not make the body of a ``WHEN``/``THEN`` take a long time to
+    execute. If you attempt to run code that lasts too long in the body
+    of your ``WHEN``/``THEN`` statement, it will cause the main
+    line code, and all other triggers (WHEN, ON, and cooked steering
+    locks) to be stuck unable to continue until it finishes.
+
+For the reason mentioned in the warning above, you also probably
+should not make the system execute a ``WAIT`` command when inside the
+body of a WHEN/THEN statement.
```

Comment:
Since this references the warning message, maybe it should be moved into the warning message itself?